src/libAtomVM/externalterm.c

```diff
@@ -211,6 +234,19 @@ static int serialize_term(Context *ctx, uint8_t *buf, term t)
         }
         return 2;
 
+    } else if (term_is_boxed_integer(t) || (term_is_integer(t) && (term_to_int(t) > INT32_MAX || term_to_int(t) < INT32_MIN))) {
```

Comment:
There is a quite interesting corner case here:
on a 32 bit machine, given that the integer tag takes 4 bits, 28 bits are available for storing an integer as a single term, so a 29 bit term is stored as a boxed integer.
However, once serialized it can be serialized as INTEGER_EXT, since that boxed number can be safely stored into a signed 32 bit integer.
Here is an example: `erlang:term_to_binary(0x7FFFFFFF)` -> `<<131, 98, 127, 255, 255, 255>>` (0x7FFFFFFF is for sure handled as a boxed integer on a 32 bit CPU).